testsuites/lbt-prelude/lbt-prelude-typescript/package-lock.json

```diff
@@ -1,72 +1,270 @@
       "dependencies": {
-        "undici-types": "~5.26.4"
+        "lbr-prelude": "file:.extra-dependencies/lbr-prelude",
+        "prelude": "file:.extra-dependencies/prelude"
+      },
+      "devDependencies": {
+        "typescript": "^5.3.3"
       }
     },
-    "node_modules/lbf-prelude": {
+    ".extra-dependencies/lbf-prelude-golden-api": {
       "version": "1.0.0",
-      "resolved": "file:.extra-dependencies/lbf-prelude-1.0.0.tgz",
-      "integrity": "sha512-jOL78xR3JwPMCiFZKxLFiaRvtpSlCNdWcp0NcekUwDQ5+Hg2Yg/nMqvB8lqg8orU+ZsWuHO4mo8MNPVMeyJavA==",
       "dependencies": {
-        "lbr-prelude": "file:.extra-dependencies/lbr-prelude-1.0.0.tgz",
-        "prelude": "file:.extra-dependencies/prelude-1.0.1.tgz"
+        "lbf-prelude": "file:.extra-dependencies/lbf-prelude",
+        "lbr-prelude": "file:.extra-dependencies/lbr-prelude",
+        "prelude": "file:.extra-dependencies/prelude"
+      },
+      "devDependencies": {
+        "typescript": "^5.3.3"
       }
     },
-    "node_modules/lbf-prelude-golden-api": {
+    ".extra-dependencies/lbf-prelude-golden-api/.extra-dependencies/lbf-prelude": {
       "version": "1.0.0",
-      "resolved": "file:.extra-dependencies/lbf-prelude-golden-api-1.0.0.tgz",
-      "integrity": "sha512-vXBgsy4ShhRkFNZDWnF6xu7sX8cSWxt9u0+9Z0SEPP3X80hZkwKi3rpsautdfe0SW8d/yx1T20q1SSWxYVBK9A==",
+      "extraneous": true,
       "dependencies": {
-        "lbf-prelude": "file:.extra-dependencies/lbf-prelude-1.0.0.tgz",
-        "lbr-prelude": "file:.extra-dependencies/lbr-prelude-1.0.0.tgz",
-        "prelude": "file:.extra-dependencies/prelude-1.0.1.tgz"
+        "lbr-prelude": "file:.extra-dependencies/lbr-prelude",
+        "prelude": "file:.extra-dependencies/prelude"
       }
     },
-    "node_modules/lbr-prelude": {
+    ".extra-dependencies/lbf-prelude-golden-api/.extra-dependencies/lbr-prelude": {
       "version": "1.0.0",
-      "resolved": "file:.extra-dependencies/lbr-prelude-1.0.0.tgz",
-      "integrity": "sha512-w9hPnbHU6np87kdChn7D7r021U0UUTDhKxc9rYZqTsu73bRfY1AxxUh4n4FgiF6aa+GFtHFaYd06a0Znka/PMQ==",
+      "extraneous": true,
       "license": "ISC",
       "dependencies": {
-        "prelude": "file:.extra-dependencies/prelude-1.0.1.tgz",
-        "typescript": "^5.3.3"
+        "prelude": "file:.extra-dependencies/prelude"
       }
     },
-    "node_modules/prelude": {
+    ".extra-dependencies/lbf-prelude-golden-api/.extra-dependencies/prelude": {
       "version": "1.0.1",
-      "resolved": "file:.extra-dependencies/prelude-1.0.1.tgz",
-      "integrity": "sha512-NtSJIsn0+eDmzHZu3o5rK5Fh2Q3/eXmZ+tcsUsip7XmKl06RX0vTWWSsYngJhsdj+Xd7jZl/Wy92m8oNycykVA==",
+      "extraneous": true,
       "license": "ISC"
     },
+    ".extra-dependencies/lbf-prelude/.extra-dependencies/lbr-prelude": {
+      "version": "1.0.0",
+      "extraneous": true,
+      "license": "ISC",
+      "dependencies": {
+        "prelude": "file:.extra-dependencies/prelude"
+      }
+    },
```

Comment:
Urgh, this is awkwardness with `npm` (yet again!). I do expect this since each package contains its transitive closure of its local dependencies (awful space complexity :/) -- the only thing that managed to convince npm that all the dependencies were there.

On the plus, these packages (with their transitive closure) are fully compatible with non nix users i.e., we can  just send these `npm` packages to other people who only use `npm` and they'll have no troubles using it.